### PR TITLE
Adding Feature to support PR Reviews

### PR DESCRIPTION
**Pull request by MIslam92**
**What is the context of this PR?**

https://trello.com/c/2cE03AG1

**How to review**
Create own token in GitHub to access new additions to Forth Wall and enter:
file:///Users/"YourName"/projects/fourth-wall/index.html?"YOUR_TOKEN"&team=onsdigital/eq

**Who worked on the PR:**
Mohammad Samiul Islam
Shahi Noor Ahmed

### CODE REVIEW

**Comment by collisdigital** [comment]
The `.idea` directory shouldn't be committed with this PR, if it's useful to have then it should be added as a separate PR

File: javascript/pull-view.js [hunk 1]
```diff
@@ -71,8 +72,12 @@
 
 
       var needsRebase = undefined;
+      var needsApproval = undefined;
+
       var baseSyncHTML = "";
-      if (this.model.branchHead.get('object') &&
+      var baseSyncHTML2= "";
```
Comment (collisdigital) [inline javascript/pull-view.js]: Is this still needed/used?

File: javascript/pull-view.js [hunk 2]
```diff
@@ -98,7 +103,7 @@
             this.secondsToTime(this.model.get('elapsed_time')),
             '</div>',
           '</div>',
-        '<div class="status-holder">', statusString , baseSyncHTML,'</div>',
+        '<div class="status-holder">', statusString , baseSyncHTML, approvalString, baseSyncHTML2, '</div>',
```
Comment (collisdigital) [inline javascript/pull-view.js]: Is `baseSyncHTML2` needed? If so name needs re-thinking

File: javascript/pull-view.js [hunk 3]
```diff
@@ -165,14 +170,44 @@
...
+      var reviewers = reviews.get('reviewers')
+
+      if (reviewers != null){
+          var totalReviews = Object.keys(reviewers).length;
+          var approved = 0;
+          var requestedChanges = 0;
+
+          for(var review in reviewers) {
+            if(reviewers[review].state == "APPROVED"){
+                approved ++;
```
Comment (collisdigital) [inline javascript/pull-view.js]: Some whitespace stuff here needs changing to match the style in the rest of the file

File: javascript/pull-view.js [hunk 4]
```diff
@@ -165,14 +170,44 @@
...
+          text = 'Approvals (' + approved + '/' + totalReviews + ')';
+      } else {
+        text = 'Unknown';
+      }
+
+      if (approved >= 2 && requestedChanges == 0 ){
+        classes = 'success';
+      }
+      else{
+        classes = 'base-sync base-sync-rebase';
```
Comment (collisdigital) [inline javascript/pull-view.js]: use unique classes for this status rather than re-using the sync ones

**Review by shahi645 — COMMENTED** [review]
Dont think all these xml files are required... CSS class needed for approval status

**Review by collisdigital — CHANGES_REQUESTED** [review]

**Comment by MIslam92** [comment]
@collisdigital I have just added them to the .gitignore file and am gonna push that.

File: javascript/pull-view.js (hunk 1 above)
Comment (MIslam92) [reply javascript/pull-view.js]: No it is redundant now in process of removing 👍

File: javascript/pull-view.js (hunk 3 above)
Comment (MIslam92) [reply javascript/pull-view.js]: Done

**Review by shahi645 — APPROVED** [review]

**Review by shahi645 — APPROVED** [review]

File: fourth-wall.css [hunk 5]
```diff
@@ -82,13 +81,28 @@ body {
...
+  background-color: green;
+  border-radius: 0px 0px 4px 4px;
+}
+
+.pending-approval {
+  background-color: orange;
+  border-radius: 0px 0px 4px 4px;
+}
+
+.needs-approval {
```
Comment (collisdigital) [inline fourth-wall.css]: this is actually `.changes-requested`

File: javascript/pull-view.js [hunk 6]
```diff
@@ -165,14 +168,51 @@
...
+
+      if (reviewers != null){
+          var totalReviews = Object.keys(reviewers).length;
+          var approved = 0;
+          var requestedChanges = 0;
+          var commented = 0;
+
+          for(var review in reviewers) {
+            if(reviewers[review].state == "APPROVED"){
+                approved ++;
```
Comment (collisdigital) [inline javascript/pull-view.js]: remove space before `++`

File: javascript/pull-view.js [hunk 7]
```diff
@@ -165,14 +168,51 @@
...
+          text = 'Approvals (' + approved + '/' + totalReviews + ')';
+      } else {
+        text = 'Unknown';
+      }
+
+      // if status is approved only if the no other statuses are present
+      if (approved >= 2 && requestedChanges == 0 && commented == 0 ){
+        classes = 'approved';
+      }
+      else if (approved < 2 || requestedChanges > 0 || commented > 0 ){
```
Comment (collisdigital) [inline javascript/pull-view.js]: if requestedChanges > 0 then it should be red. Consider putting that statement first e.g.

```
if (requestedChanges > 0) {
    classes = 'changes-requested';
}
else if (approved < 2) {
    classes = 'pending-approval';
}
else {
    classes = 'approved';
}
```

**Review by collisdigital — CHANGES_REQUESTED** [review]

File: javascript/pull-view.js (hunk 7 above)
Comment (MIslam92) [reply javascript/pull-view.js]: @collisdigital All changes made as per request 👍

**Review by collisdigital — APPROVED** [review]

**Comment by collisdigital** [comment]
Please squash then this can merge

**Pull request merged**